app/src/utils/data/grantRounds.ts

```diff
@@ -145,103 +145,115 @@ export async function getGrantRound(blockNumber: number, grantRoundAddress: stri
-      if (forceRefresh || !LocalForageData) {
-        // Define calls to be read using multicall
-        [
-          donationTokenAddress,
-          startTime,
-          endTime,
-          metadataAdmin,
-          payoutAdmin,
-          registryAddress,
-          metaPtr,
-          hasPaidOut,
-          funds,
-        ] = await callMulticallContract([
-          // pull the grantRound data from its contract
-          {
-            target: grantRoundAddress,
-            contract: roundContract,
-            fns: [
-              'donationToken',
-              'startTime',
-              'endTime',
-              'metadataAdmin',
-              'payoutAdmin',
-              'registry',
-              'metaPtr',
-              'hasPaidOut',
-            ],
-          },
-          // get the balance from the matchinTokenContract
-          {
-            target: matchingTokenAddress,
-            contract: matchingTokenContract,
-            fns: [
-              {
-                fn: 'balanceOf',
-                args: [grantRoundAddress],
-              },
-            ],
-          },
-        ]);
-        // get the donation/matching token
-        matchingToken = SUPPORTED_TOKENS_MAPPING[matchingTokenAddress];
-        donationToken = SUPPORTED_TOKENS_MAPPING[donationTokenAddress];
-        // record the funds as a human readable number
-        funds = parseFloat(formatUnits(BigNumber.from(funds), SUPPORTED_TOKENS_MAPPING[matchingTokenAddress].decimals));
-      } else if (LocalForageData && _lsBlockNumber < blockNumber) {
-        // get updated metadata
-        const [newMetaPtr, balance] = await Promise.all([
-          roundContract.metaPtr(),
-          matchingTokenContract.balanceOf(grantRoundAddress),
-        ]);
-        // update to the new metaPtr
-        metaPtr = newMetaPtr;
-        // update to the new balance
-        funds = parseFloat(formatUnits(balance, SUPPORTED_TOKENS_MAPPING[matchingTokenAddress].decimals)).toString();
-      }
-      // build status against now (unix)
-      const now = Date.now() / 1000;
-      // place the GrantRound details into a GrantRound object
-      const grantRound = {
-        grantRound: {
-          startTime,
-          endTime,
-          metadataAdmin,
-          payoutAdmin,
-          registryAddress,
-          metaPtr,
-          hasPaidOut,
-          donationToken: donationToken,
-          matchingToken: matchingToken,
-          address: grantRoundAddress,
-          funds: funds,
-          status:
-            now >= BigNumber.from(startTime).toNumber() && now < BigNumber.from(endTime).toNumber()
-              ? 'Active'
-              : now < BigNumber.from(startTime).toNumber()
-              ? 'Upcoming'
-              : 'Completed',
-          registry: GRANT_REGISTRY_ADDRESS,
-          error: undefined,
-        } as GrantRound,
-      };
+      // no contract deployed here...
+      if ((await provider.value.getCode(grantRoundAddress)) === '0x') {
```

Comment:
When the `DEFAULT_PROVIDER` connects on `rawInit`, it will be pointing at the correct chain, so that check will pass, then here we're checking if the `GrantRound`s we've pulled from storage are deployed on this network. 

We could probably improve the `rawInit` check to avoid this if we stored the `GRANT_REGISTRY_ADDRESS` and `GRANT_MANAGER_ADDRESS` in to indexedDB then if those are ever out of sync, we could clear all of the cached entries?